include/seastar/util/program-options.hh

```diff
@@ -463,6 +466,61 @@ public:
     void set_value(T value) { do_set_value(std::move(value), false); }
 };
 
+template <typename T = std::monostate>
+class updateable_value : public value<T> {
+    std::optional<T> _default_value;
+    bool _locked = false;
+
+public:
+    class lock {
+        updateable_value* _v;
+
+    public:
+        lock(updateable_value& v) : _v(&v) {
+            _v->_locked = true;
+        }
+        ~lock() {
+            if (_v != nullptr) {
+                _v->_locked = false;
+            }
+        }
+        lock(const lock&) = delete;
+        lock(lock&& o) noexcept : _v(std::exchange(o._v, nullptr)) {}
+    };
+
+private:
+    noncopyable_function<void(const updateable_value&, lock)> _update;
+
+    virtual void do_mutate_live(options_mutator& mutator) override {
+        if (_locked) {
+            return;
+        }
+
+        T val;
+        if (mutator.visit_value(val)) {
+            value<T>::do_set_value(std::move(val), false);
+            _update(*this, lock(*this));
+        } else if (!value<T>::defaulted()) {
+            value<T>::do_set_value(*_default_value, true);
```

Comment:
This will fail if `_default_value` is `nullopt`. You should add a `do_set_value()` overload which takes an optional, so you can pass `_default_value` as-is.

Reply:
Yikes, indeed